features/dd-sdk-android-session-replay/src/test/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/BitmapPoolTest.kt

```diff
@@ -358,6 +362,40 @@ internal class BitmapPoolTest {
         assertThat(captor.firstValue).isEqualTo(0)
     }
 
+    @RepeatedTest(30)
+    fun `M not receive ConcurrentModificationException W get() { and onLowMemory }`(
+        forge: Forge
+    ) {
+        // when this issue occurs, the frequency is roughly once per 3000 runs,
+
+        // Given
+        val numberOfThreads = 3
+        val executor = Executors.newFixedThreadPool(numberOfThreads)
+        val tasks = mutableListOf<Future<*>>()
+
+        val allThreadsStartedLatch = CountDownLatch(numberOfThreads)
+        testedCache.put(mockBitmap)
+        val bitmapKey = spyBitmapPoolHelper.generateKey(mockBitmap)
+
+        // When
+        repeat(numberOfThreads) {
+            tasks += executor.submit {
+                if (forge.aBool()) {
+                    allThreadsStartedLatch.countDown()
+                    allThreadsStartedLatch.await()
+                    testedCache.get(bitmapKey)
+                } else {
+                    allThreadsStartedLatch.countDown()
+                    allThreadsStartedLatch.await()
+                    testedCache.onLowMemory()
+                }
```

Comment:
common part can be lifted out.

```suggestion
                allThreadsStartedLatch.countDown()
                allThreadsStartedLatch.await()
                if (forge.aBool()) {
                    testedCache.get(bitmapKey)
                } else {
                    testedCache.onLowMemory()
                }
```